src-core/src/generation/generate.rs

```diff
@@ -125,5 +128,11 @@ pub fn generate(
     gen.add_omni_transformer::<ConstraintSetter>();
     gen.add_omni_transformer::<CallbackSetter>();
 
-    gen.generate()
+    let new_traj = gen.generate();
+    if new_traj.is_ok() {
+        original.trajectory = new_traj?.trajectory;
+        Ok(original)
+    } else {
+        new_traj
+    }
```

Comment:
This should be equivalent, but more Rust-y.
```suggestion
    match gen.generate() {
        Err(e) => Err(e),
        Ok(new_trajectory_file) => {
            original_file.trajectory = new_trajectory_file.trajectory;
            Ok(original_file)
        }
    }
```
Why does returning the original file instead of the new file here fix the problem?